devcycle_python_sdk/dvc_local_client.py

```diff
@@ -30,61 +30,54 @@ def __init__(self, sdk_key: str, options: DevCycleLocalOptions):
         self.local_bucketing = LocalBucketing(sdk_key)
 
         self._platform_data = default_platform_data()
-        self.local_bucketing.set_platform_data(json.dumps(self._platform_data.to_json()))
-
-        self.config_manager: EnvironmentConfigManager = EnvironmentConfigManager(sdk_key, self.options,
-                                                                                 self.local_bucketing)
-        self.event_queue_manager: EventQueueManager = EventQueueManager(sdk_key, self.options, self.local_bucketing)
-
-    def _add_platform_data_to_user(self, user: User) -> User:
-        user.platform = self._platform_data.platform
-        user.platformVersion = self._platform_data.platformVersion
-        user.sdkVersion = self._platform_data.sdkVersion
-        user.sdkType = self._platform_data.sdkType
-        return user
-
-    def _validate_sdk_key(self, sdk_key: str) -> None:
-        if sdk_key is None or len(sdk_key) == 0:
-            raise ValueError("Missing SDK key! Call build with a valid server SDK key")
-
-        if not sdk_key.startswith("server") and not sdk_key.startswith("dvc_server"):
-            raise ValueError(
-                "Invalid SDK key provided. Call build with a valid server SDK key"
-            )
-
-    def _validate_user(self, user: User) -> None:
-        if user is None:
-            raise ValueError("User cannot be None")
+        self.local_bucketing.set_platform_data(
+            json.dumps(self._platform_data.to_json())
+        )
 
-        if user.user_id is None or len(user.user_id) == 0:
-            raise ValueError("userId cannot be empty")
+        self.config_manager: EnvironmentConfigManager = EnvironmentConfigManager(
+            sdk_key, self.options, self.local_bucketing
+        )
+        self.event_queue_manager: EventQueueManager = EventQueueManager(
+            sdk_key, self.options, self.local_bucketing
+        )
 
-    def _is_initialized(self) -> bool:
-        return (self.config_manager and self.config_manager.is_initialized())
+    def is_initialized(self) -> bool:
+        return self.config_manager and self.config_manager.is_initialized()
 
     def set_client_custom_data(self, custom_data: Dict[str, Any]) -> None:
-        if not self._is_initialized():
+        """
+        Sets global custom data for this client. This data will be utilized in all segmentation and bucketing
+        decisions. This data will be merged with any custom data set on the user object, with user data
+        taking priority
+
+        :param custom_data: Global data to set.  Supported values are strings, numbers, booleans, and lists of strings,
```

Comment:
I think we could put this requirement into the parameter type above?
Up to you whether you think it's worth it.

Reply:
i pushed a change to (a) fix the bad comment and (b) try out some more type hinting for the dictionary values